ha/core/controllers/pcs/node_controller.py

```diff
@@ -280,31 +301,30 @@ def stop(self, node_id: str, timeout: int= -1, **op_kwargs) -> dict:
             ([dict]): Return dictionary. {"status": "", "output": "", "error": ""}
                 status: Succeeded, Failed, InProgress
         """
-        # Get the node_name (pvtfqdn) fron nodeid
+        # Get the node_name (pvtfqdn) fron nodeid and raise exception if node_id is not valid
         node_name = self._get_node_name(node_id=node_id)
-        # TODO: check_cluster - boolean.It checks whether cluster is going to be offline if node with node_id is stopped.
-        timeout = const.NODE_STOP_TIMEOUT if timeout < 0 else timeout
-        # TODO: Get the node_status from system health status
-        node_status = self.nodes_status([node_name]).get(node_name)
-        if node_status == NODE_STATUSES.CLUSTER_OFFLINE.value:
-            Log.info(f"For stop {node_name}, Node already in offline state.")
-            status = f"Node {node_name} is already in offline state."
-        elif node_status == NODE_STATUSES.POWEROFF.value:
-            raise ClusterManagerError(f"Failed to stop {node_name}. Node is in {node_status}.")
-        else:
-            if self.heal_resource(node_name):
-                time.sleep(const.BASE_WAIT_TIME)
-            try:
-                Log.info(f"Please Wait, trying to stop node: {node_name}")
-                # TODO: Use PCS_STOP_NODE from const.py with timeout value
-                self._execute.run_cmd(f"pcs cluster stop {node_name}")
-                Log.info(f"Executed node stop for {node_name}, Waiting to stop resource")
-                time.sleep(const.BASE_WAIT_TIME)
-                status = f"Stop for {node_name} is in progress, waiting to stop resource"
-                # TODO: Update node_status in system_health
-            except Exception as e:
+        try:
+            stop_status = json.loads(super().stop(node_id, **op_kwargs))
+            if stop_status != None:
+                if stop_status["status"] == const.STATUSES.SUCCEEDED.value or stop_status["status"] == const.STATUSES.FAILED.value:
+                    return stop_status
+            Log.info(f"Please Wait, trying to stop node: {node_name}")
+            # TODO: Use PCS_STOP_NODE from const.py with timeout value
```

Comment:
Please raise a improvement ticket for this TODO so that it won't be missed.

Reply:
Added ticket id in comment.